paasta_tools/list_marathon_service_instances.py

```diff
@@ -25,34 +25,114 @@
 
 - -d <SOA_DIR>, --soa-dir <SOA_DIR>: Specify a SOA config dir to read from
 - -c <CLUSTER>, --cluster <CLUSTER>: Specify which cluster of services to read
+- -m, --minimal: Only show service instances that need bouncing
 """
 import argparse
 import sys
 
 from paasta_tools.marathon_tools import DEFAULT_SOA_DIR
+from paasta_tools.marathon_tools import get_marathon_client
+from paasta_tools.marathon_tools import get_num_at_risk_tasks
+from paasta_tools.marathon_tools import load_marathon_config
+from paasta_tools.marathon_tools import load_marathon_service_config
 from paasta_tools.utils import compose_job_id
+from paasta_tools.utils import decompose_job_id
 from paasta_tools.utils import get_services_for_cluster
+from paasta_tools.utils import load_system_paasta_config
+from paasta_tools.utils import NoDockerImageError
+from paasta_tools.utils import use_requests_cache
 
 
 def parse_args():
-    parser = argparse.ArgumentParser(description='Lists marathon instances for a service.')
+    parser = argparse.ArgumentParser(
+        description='Lists marathon instances for a service.')
     parser.add_argument('-c', '--cluster', dest="cluster", metavar="CLUSTER",
                         default=None,
                         help="define a specific cluster to read from")
     parser.add_argument('-d', '--soa-dir', dest="soa_dir", metavar="SOA_DIR",
                         default=DEFAULT_SOA_DIR,
                         help="define a different soa config directory")
+    parser.add_argument('-m', '--minimal', dest='minimal', action='store_true',
+                        help="show only service instances that need bouncing")
     args = parser.parse_args()
     return args
 
 
+def long_job_id_to_short_job_id(long_job_id):
```

Comment:
Services that are deleted from soa-configs will be marked as needing bouncing.

I didn't want to use `get_matching_apps` since I don't want the fuzzy matching. Also `get_matching_apps` and `get_matching_appids` go from short to long, which is the opposite of what I want.